include/ros_alarms/listener.hpp

```diff
@@ -54,115 +59,148 @@ class AlarmListener
-  bool query_raised()           { get_alarm(); return __last_alarm.raised; }
-  bool query_cleared()          { get_alarm(); return !is_raised(); }
+  bool queryRaised()           { getAlarm(); return __last_alarm.raised; }
+  bool queryCleared()          { getAlarm(); return !isRaised(); }
 
   // Queries server and parses response into an AlarmProxy
-  AlarmProxy get_alarm();
+  AlarmProxy getAlarm();
 
   // Returns AlarmProxy for last alarm of this name published to '/alarm/updates'
-  AlarmProxy get_cached_alarm() { return __last_alarm; }
+  AlarmProxy getCachedAlarm() { return __last_alarm; }
 
   // Registers a callback to be invoked on both raise and clear
-  void add_cb(callable_t cb);
+  void addCb(callable_t cb);
 
   // Registers a raise callback to be called for any severity level
-  void add_raise_cb(callable_t cb);
+  void addRaiseCb(callable_t cb);
 
   // Registers a raise callback to be called for a single severity level
-  void add_raise_cb(callable_t cb, int severity);
+  void addRaiseCb(callable_t cb, int severity);
 
   // Registers a callback to be invoked at alarm raise for a range of severities
-  void add_raise_cb(callable_t cb, int severity_lo, int severity_hi);
+  void addRaiseCb(callable_t cb, int severity_lo, int severity_hi);
 
   // Registers a callback to be invoked when alarm is cleared
-  void add_clear_cb(callable_t cb);
+  void addClearCb(callable_t cb);
 
   // Returns the last time the alarm was updated
-  ros::Time last_update_time() const { return __last_update; }
+  ros::Time getLastUpdateTime() const { return __last_update; }
 
   // Returns the time since last update
-  ros::Duration time_since_update() const { return ros::Time::now() - __last_update; }
+  ros::Duration getTimeSinceUpdate() const { return ros::Time::now() - __last_update; }
+
+  // Waits for an update to the alarm via '/alarm/udates' with timeout
+  bool waitForUpdate(ros::Duration timeout = ros::Duration(-1.0)) const;
 
-  void clear_callbacks()      { __callbacks.clear(); }
+  void clearCallbacks() { __callbacks.clear(); }
 
 private:
   bool __ok = true;  // listener fail-bit
   ros::NodeHandle __nh;
   std::string __alarm_name;
   ros::ServiceClient __get_alarm;
+  ros::CallbackQueue __cb_queue;
+  ros::AsyncSpinner __async_spinner;
   ros::Subscriber __update_subscriber;
   std::vector<ListenerCb<callable_t>> __callbacks;
   AlarmProxy __last_alarm;
   ros::Time __last_update { 0, 0 };
-  void __add_cb(callable_t cb, int severity_lo, int severity_hi, CallScenario call_scenario);
-  void __alarm_update(ros_alarms::Alarm);
+  std::string __object_name;
+  void __addCb(callable_t cb, int severity_lo, int severity_hi,
+               CallScenario call_scenario);
+  void __alarmUpdate(ros_alarms::Alarm);
 };
 
 
 template <typename callable_t>
 AlarmListener<callable_t>
 ::AlarmListener(ros::NodeHandle &nh, std::string alarm_name)
-: __nh(nh), __alarm_name(alarm_name)
+try
```

Comment:
This is the syntax that is used if you want to catch exceptions thrown from initializers. The catch block directly follows the ctor. This is still just calling the objects respective constructors as ususal.